lib/logstash/outputs/cloudwatchlogs.rb

```diff
@@ -219,7 +246,7 @@ def put_log_events(log_events)
 
   private
   def invalid?(event)
-    status = event[TIMESTAMP].nil? || event[MESSAGE].nil?
+    status = event[TIMESTAMP].nil? || (!@use_codec && event[MESSAGE].nil?)
     if status
```

Comment:
Nit: `@logger.warn("Skipping invalid event #{event.to_hash}") if status` is more idiomatic.